lib/asset/uploadAnimation.js

```diff
@@ -0,0 +1,103 @@
+ * @param {string | Stream} data - The rbxm file containing the KeyframeSequence.
+ * @param {object=} itemOptions - The options for the upload. Only optional if assetId is not provided.
+ * @param {string=} itemOptions.name - The name of the animation.
+ * @param {string=} itemOptions.description - The description for the animation.
+ * @param {boolean=} itemOptions.copyLocked - If the animation is copy-locked.
+ * @param {boolean=} itemOptions.allowComments - If comments are allowed.
+ * @param {number=} itemOptions.groupId - The group to upload the animation to. This is ignored if the assetId is provided.
+ * @param {number=} assetId - An existing assetId to overwrite.
+ * @returns {Promise<number>}
+ * @example const noblox = require("noblox.js")
+ * const fs = require("fs")
+ * // Login using your cookie
+ * const assetId = await noblox.uploadAnimation(fs.readFileSync("./KeyframeSequence.rbxm"), {
+ *  name: "A cool animation",
+ *  description: "This is a very cool animation",
+ *  copyLocked: false, //The asset is allowed to be copied.
+ *  allowComments: false
+ * }, 7132858975)
+**/
+
+// Define
+function upload (data, itemOptions, assetId, jar, token) {
+  const httpOpt = {
+    url: assetId ? `//www.roblox.com/ide/publish/uploadexistinganimation?assetID=${assetId}&isGamesAsset=False` : '//www.roblox.com/ide/publish/uploadnewanimation?AllID=1',
+    options: {
+      resolveWithFullResponse: true,
+      method: 'POST',
+      jar: jar,
+      body: data,
+      headers: {
+        'X-CSRF-TOKEN': token,
+        'Content-Type': 'application/xml',
+        'User-Agent': 'RobloxStudio/WinInet RobloxApp/0.483.1.425021 (GlobalDist; RobloxDirectDownload)'
+      }
+    }
+  }
+
+  if (!assetId && itemOptions) {
+    const copyLocked = itemOptions.copyLocked
+    const allowComments = itemOptions.allowComments
+    httpOpt.url += '&assetTypeName=Animation&genreTypeId=1&name=' +
+      itemOptions.name +
+      '&description=' +
+      (itemOptions.description || '') +
+      '&ispublic=' +
+      (copyLocked != null ? !copyLocked : false) +
+      '&allowComments=' +
+      (allowComments != null ? allowComments : true) +
+      '&groupId=' +
+      (itemOptions.groupId || '')
+  } else if (!assetId) {
+    throw new Error('ItemOptions is required for new assets.')
+  }
+
+  return http(httpOpt)
+    .then(function (res) {
+      if (res.statusCode === 200) {
+        const resultId = assetId || Number(res.body)
+
+        if (assetId && itemOptions) {
+          const copyLocked = itemOptions.copyLocked
+          const allowComments = itemOptions.allowComments
+
+          return configureItem({
+            id: resultId,
+            name: itemOptions.name,
+            description: itemOptions.description,
+            enableComments: (allowComments != null ? allowComments : true),
+            sellForRobux: (copyLocked != null ? !copyLocked : false)
+          }).then(function () {
+            return resultId
```

Comment:
I kept out a sub-request as it would just be extra bloat in the library for a simple request they could do themselves in 1 line of code if really needed. Also, the rate limits are obviously not the same so I'm not sure how those two would interact if someone is doing mass uploading. That's just my opinion on the matter.